### PR TITLE
finish_cotuning

### PR DESCRIPTION
**Pull request by Tongjiaxun**
finish label-cotuning

### CODE REVIEW

File: configs/faster_rcnn/faster_rcnn_r50_vd_fpn_1x_cotuning_coco.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+_BASE_: [
```
Comment (jerrywgz) [inline configs/faster_rcnn/faster_rcnn_r50_vd_fpn_1x_cotuning_coco.yml]: 建议文件名改为coco_cotuning_roadsign后缀

File: ppdet/modeling/heads/bbox_head.py [hunk 2]
```diff
@@ -185,7 +185,10 @@ def __init__(self,
                  num_classes=80,
                  bbox_weight=[10., 10., 5., 5.],
                  bbox_loss=None,
-                 loss_normalize_pos=False):
+                 loss_normalize_pos=False,
+                 cot_classes=None,
```
Comment (jerrywgz) [inline ppdet/modeling/heads/bbox_head.py:192]: 需要增加注释，说明配置方法

File: ppdet/modeling/losses/cot_loss.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    __shared__ = ['num_classes']
+    def __init__(self,
+                 num_classes=80, 
+                 cot_scale=1,
+                 cot_lambda=1):
+        super(COTLoss, self).__init__()
+        self.cot_scale = cot_scale
+        self.cot_lambda = cot_lambda    
+        self.num_classes = num_classes    
+        print('cot_scale:{x}'.format(x=self.cot_scale))
```
Comment (jerrywgz) [inline ppdet/modeling/losses/cot_loss.py]: 去除这部分打印

File: tools/train.py [hunk 4]
```diff
@@ -125,7 +125,10 @@ def run(FLAGS, cfg):
         set_random_seed(0)
 
     # build trainer
-    trainer = Trainer(cfg, mode='train')
+    if cfg.use_cot:
+        trainer = TrainerCot(cfg, mode='train')
```
Comment (jerrywgz) [inline tools/train.py:140]: 使用该trainer，是否验证过模型导出和部署

File: ppdet/engine/trainercot.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (jerrywgz) [inline ppdet/engine/trainercot.py]: 2020->2022

File: ppdet/engine/trainercot.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+from .callbacks import Callback, ComposeCallback, LogPrinter, Checkpointer, WiferFaceEval, VisualDLWriter, SniperProposalsGenerator
+from .export_utils import _dump_infer_config, _prune_input_spec
+from ppdet.utils.logger import setup_logger
+logger = setup_logger('ppdet.engine')
+
+from . import Trainer
+__all__ = ['TrainerCot']
+
+class TrainerCot(Trainer):
```
Comment (jerrywgz) [inline ppdet/engine/trainercot.py:52]: 补充下注释说明该trainer的目的

File: configs/faster_rcnn/faster_rcnn_r50_vd_fpn_1x_coco_cotuning_roadsign.yml [hunk 7]
```diff
@@ -0,0 +1,63 @@
+_BASE_: [
```
Comment (jerrywgz) [inline configs/faster_rcnn/faster_rcnn_r50_vd_fpn_1x_coco_cotuning_roadsign.yml]: 配置文件新建一个cotuning的文件夹吧，然后faster rcnn的这份配置放到label cotuning里面，再补充下文档，里面包括参考的文章还有模型效果的表格，可以参考configs/faster_rcnn下的文档

File: configs/cotuning/_base_/faster_rcnn_swin_reader.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
+worker_num: 2
```
Comment (jerrywgz) [inline configs/cotuning/_base_/faster_rcnn_swin_reader.yml]: 不需要swin的配置吧

**Review by jerrywgz — DISMISSED** [review]

**Comment by nemonameless** [comment]
#7503  merged
